src/Relation/Binary/Construct/Interior/Symmetric.agda

```diff
@@ -0,0 +1,91 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Symmetric interior of a binary relation
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Relation.Binary.Construct.Interior.Symmetric where
+
+open import Function.Base using (flip)
+open import Level
+open import Relation.Binary
+
+private
+  variable
+    a b c ℓ r s t : Level
+    A : Set a
+    R : Rel A r
+    S : Rel A s
+    T : Rel A t
+
+------------------------------------------------------------------------
+-- Definition
+
+record SymInterior (R : Rel A ℓ) (x y : A) : Set ℓ where
+  constructor _,_
+  field
+    lhs≤rhs : R x y
+    rhs≤lhs : R y x
+open SymInterior public
+
+------------------------------------------------------------------------
+-- Properties
+
+-- The symmetric interior is symmetric.
+symmetric : Symmetric (SymInterior R)
+symmetric (r , r′) = r′ , r
+
+-- The symmetric interior of R is greater than (or equal to) any other symmetric
+-- relation contained by R.
+unfold : Symmetric S → S ⇒ R → S ⇒ SymInterior R
+unfold sym f s = f s , f (sym s)
+
+-- SymInterior preserves various properties.
+reflexive : Reflexive R → Reflexive (SymInterior R)
+reflexive refl = refl , refl
+
+transitive′ : Trans R S T → Trans S R T →
+  Trans (SymInterior R) (SymInterior S) (SymInterior T)
+transitive′ trans trans′ (r , r′) (s , s′) = trans r s , trans′ s′ r′
+
+transitive : Transitive R → Transitive (SymInterior R)
+transitive {R = R} tr = transitive′ {R = R} tr tr
+
+-- The symmetric interior of a strict relation is empty.
+Empty-SymInterior : Asymmetric R → Empty (SymInterior R)
+Empty-SymInterior asym (r , r′) = asym r r′
+
+-- A reflexive transitive relation _≤_ gives rise to a poset in which the
+-- equivalence relation is SymInterior _≤_.
+record IsProset {a ℓ} {A : Set a} (≤ : Rel A ℓ) : Set (a ⊔ ℓ) where
+  field
+    refl : Reflexive ≤
+    trans : Transitive ≤
```

Comment:
Conventionally, given what you subsequent define as the bundled contents, better to declare the relation argument here also as `_≤_`?

... and even in L26-L30 above?

Reply:
I was remembering the convention from `Algebra.Structures`, which is different to that of `Relation.Binary.Structures`. I find `_≤_` more readable, to be honest, so I'll switch to it here. I don't want to use it in `SymInterior` because it might not be an order relation (not being reflexive or transitive in general).